sphinx/domains/python.py

```diff
@@ -679,7 +679,29 @@ def get_index_text(self, modname: str, name_cls: Tuple[str, str]) -> str:
             else:
                 return name
 
-        return _('%s (%s attribute)') % (attrname, clsname)
+        return self.fmt % (attrname, clsname)
+
+
+class PyProperty(PyAttribute):
+    """Description of a property."""
+
+    option_spec = PyAttribute.option_spec.copy()
+    option_spec.update({
+        'abstractmethod': directives.flag,
+    })
+    fmt = _('%s (%s property)')
+
+    def run(self) -> List[Node]:
+        # properties and attributes need to be transparently switchable:
+        # making one into the other doesn’t break APIs so it shouldn’t break doc links.
```

Comment:
What API you meant? -1 for replacing.

Reply:
That’s the main point of the whole PR, and I explained it several times so far.

Developers write a class with an attribute. Then they change the attribute to a property. People using that class won’t feel any kind of difference, the change is completely transparent.

The same should apply to Sphinx. Anyone who links an attribute or property shouldn’t ever have to change the link, no matter if the thing they link to switches from attribute to property or the other way round. Therefore the `objects.inv` category and link role for attribute and class should be the same. Since the `py:attribute` category and `attr` role already exist, I chose to use those (instead of e.g. creating a new `py:attrprop` category or so).

Reply:
As I commented in this PR several times so far, property is not a kind of attribute. So -1 to give `py:attribute` for properties. I think it is better to give a new category `py:property` to properties and make `:py:attr:` role referencible for it like `:py:meth:` does now.

I agreed with allowing to refer properties via `:py:attr:` role. But I did not say they are same as attribute. If so, no reason to add `py:property` directive.

Reply:
I’m OK with that, thank you for the clarification.